applications/HDF5Application/python_scripts/core/operations/model_part.py

```diff
@@ -16,6 +16,8 @@ def Prefix(pattern, model_part, time_format=''):
     if hasattr(model_part, 'ProcessInfo'):
         time = model_part.ProcessInfo[KratosMultiphysics.TIME]
         prefix = format(time, time_format).join(pattern.split('<time>'))
+        if KratosMultiphysics.STEP in model_part.ProcessInfo:
+            prefix = prefix.replace('<step>', str(model_part.ProcessInfo[KratosMultiphysics.STEP]))
```

Comment:
Not to change right now, writing down the idea just to not forget it, but if we are going to stick with this I would change it to something more generic (take it as an example):

```python
process_info_tokens = [
    {'name':'<time>', 'value_src':KratosMultiphysics.TIME, 'action': (lambda x: format(x, time_format))},
    {'name':'<step>', 'value_src':KratosMultiphysics.STEP, 'action': (lambda x: x)}
] # value_src could come from ProjectParameters as well, time_format->pattern_format

prefix = pattern

if hasattr(model_part, 'ProcessInfo'):
    for token in process_info_tokens:
        if token['name'] in prefix:
            if token['value_src'] in model_part.ProcessInfo:
                token_value_string = token['action'](model_part.ProcessInfo[token['value_src']])
            else:
                token_value_string = "UNDEFINED" # Maybe use a random name with the info of the case inside to identify the result?
                print("Warning: Trying to replace token {} with non existing variable {}. Using {} as dataset name.".format(
                    token['name'],
                    token['value_src'].Name(),
                    token_value_string
                ))
            prefix = prefix.replace(token['name'], str(token_value_string))
if hasattr(model_part, 'Name'):
    prefix = prefix.replace('<model_part_name>', model_part.Name)
return prefix
```

This would prevent a crash if you are trying to print using the <step> in a pattern before the STEP variable has been added to the ProcessInfo (happened to me this morning). That also would allow to easily extend the token system in the ProjectParameters to virtually use any variable we want.